mentat/utils/path.py

```diff
@@ -0,0 +1,160 @@
+import fnmatch
+import os
+import re
+from pathlib import Path
+from typing import Iterable, List, Set
+
+from ipdb import set_trace
+
+from mentat.errors import PathValidationException
+from mentat.git_handler import check_is_git_repo, get_non_gitignored_files
+
+
+# TODO: replace this with something that doesn't load the file into memory
+def is_file_text_encoded(abs_path: Path):
+    """Checks if a file is text encoded."""
+    try:
+        # The ultimate filetype test
+        with open(abs_path, "r") as f:
+            f.read()
+        return True
+    except UnicodeDecodeError:
+        return False
+
+
+def validate_and_format_path(path: Path, cwd: Path, check_for_text: bool = True) -> Path:
+    """Validate and format a path.
+
+    Args:
+        `path` - A file path, file interval path, directory path, or a glob pattern
+        `check_for_text` - Check if the file can be opened. Default to True
+
+    Return:
+        An absolute path
+    """
+    # Get absolute path
+    if path.is_absolute():
+        abs_path = path
+    else:
+        abs_path = cwd.joinpath(path)
+
+    # Validate different path types
+    # File
+    if abs_path.is_file():
+        if not abs_path.exists():
+            raise PathValidationException(f"File {abs_path} does not exist")
+        if check_for_text and not is_file_text_encoded(abs_path):
+            raise PathValidationException(f"Unable to read file {abs_path}")
+    # File interval
+    elif ":" in str(abs_path):
```

Comment:
Similar to @jakethekoenig's other comment we can't just check for a colon; we need to put this after the check for dirs and glob patterns (and also fix a few bugs in this elif statement; make sure we split the last colon -- in case the filename has a colon -- and that we actually check that the interval is valid, since otherwise we might split on a path like file/with:colon/filename and check for file/with which might actually exist). I feel like we should make a class function in Interval that does this all automatically.